src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs

```
@@ -405,24 +406,33 @@ public async Task<IActionResult> EditUserDetails(EditUserDetailsViewModel editUs
 
         // need to temporarily save the details for the next page, without saving to the database
         // however this bit throws an exception at the moment for some reason
-        //TempData.Add("NewUserDetails", editUserDetailsViewModel);
-        
+        // TempData.Add("NewUserDetails", editUserDetailsViewModel);
+        TempData.Add("NewUserDetails", System.Text.Json.JsonSerializer.Serialize(editUserDetailsViewModel));
         return RedirectToAction("CheckYourDetails", editUserDetailsViewModel);
     }
     [HttpGet]
     [Route(PagePath.CheckYourDetails)]
-    public async Task<IActionResult> CheckYourDetails( )
+    public async Task<IActionResult> CheckYourDetails()
     {
         var session = await _sessionManager.GetSessionAsync(HttpContext.Session);
         var userData = User.GetUserData();
 
+        var editUserDetailsViewModel = new EditUserDetailsViewModel();
+
+        try
+        {
+            editUserDetailsViewModel = System.Text.Json.JsonSerializer.Deserialize<EditUserDetailsViewModel>(TempData["NewUserDetails"] as string);
+        }
+        catch (Exception) { }
```

We shouldn't have an empty catch - what happens when this gets executed. IF it is right, and works like this, there should at least be a comment as to why.